pkg/apis/velero/v1/backup.go

```diff
@@ -107,14 +133,22 @@ type BackupResourceHook struct {
 type ExecHook struct {
 	// Container is the container in the pod where the command should be executed. If not specified,
 	// the pod's first container is used.
-	Container string `json:"container"`
+	// +optional
+	Container string `json:"container,omitempty"`
+
 	// Command is the command and arguments to execute.
+	// +kubebuilder:validation:MinItems=1
 	Command []string `json:"command"`
+
 	// OnError specifies how Velero should behave if it encounters an error executing this hook.
-	OnError HookErrorMode `json:"onError"`
+	// +optional
+	// +kubebuilder:validation:Enum=Continue;Fail
```

Comment:
yuck, is this the only way to handle enums?

Reply:
Unfortunately yeah :/, controller-tools isn't able to pick this up from the constants we've defined. I can move this annotation down to where `HookErrorMode` is defined, as in the example [here](https://book.kubebuilder.io/reference/generating-crd.html#validation), maybe that will help make sure we don't miss it if we add an extra value in the future?

Reply:
yeah, I think moving it might be better so it's right next to where the values are.